### PR TITLE
{Microsoft Entra ID} `az ad sp create-for-rbac`: Update examples and warnings

### PR DESCRIPTION
**Pull request by jiasli**
**Related command**
`az ad sp create-for-rbac`

**Description**

- Add examples for using `--cert` to clarify https://github.com/MicrosoftDocs/azure-docs-cli/issues/4734
- Change warning messages for `--create-cert` according to https://github.com/Azure/azure-cli/pull/30091

**Testing Guide**
`az ad sp create-for-rbac -h`

### CODE REVIEW

**Comment by yonzhan** [comment]
Thank you for your contribution! We will review the pull request and get back to you soon.

File: src/azure-cli/azure/cli/command_modules/role/_help.py [hunk 1]
```diff
@@ -528,18 +528,27 @@
...
+  text: az ad sp create-for-rbac --create-cert
+- name: Create using an existing certificate string.
+  text: az ad sp create-for-rbac --cert "MIICoT..."
+- name: Create using an existing certificate file.
+  text: |-
+      az ad sp create-for-rbac --cert "@~/cert.pem"
+      `cert.pem` contains the following content
+      -----BEGIN CERTIFICATE-----  <<< this line is optional
+      MIICoT...
+      -----END CERTIFICATE-----    <<< this line is optional
```
Comment (jiasli) [inline src/azure-cli/azure/cli/command_modules/role/_help.py:547]: These examples are similar to 

https://github.com/Azure/azure-cli/blob/031de0db795abf175cd7de7b3d17d4e8a0108b2b/src/azure-cli/azure/cli/command_modules/role/_help.py#L127-L135

https://github.com/Azure/azure-cli/blob/031de0db795abf175cd7de7b3d17d4e8a0108b2b/src/azure-cli/azure/cli/command_modules/role/_help.py#L594-L602

File: src/azure-cli/azure/cli/command_modules/role/_help.py [hunk 2]
```diff
@@ -528,18 +528,27 @@
...
+- name: Create using a self-signed certificate, and store it within Azure Key Vault.
+  text: az ad sp create-for-rbac --keyvault MyVault --cert CertName --create-cert
+- name: Create using existing certificate in Azure Key Vault.
+  text: az ad sp create-for-rbac --keyvault MyVault --cert CertName
```
Comment (jiasli) [inline src/azure-cli/azure/cli/command_modules/role/_help.py:551]: The official name should be "Azure Key Vault", not "KeyVault": https://learn.microsoft.com/en-us/azure/key-vault/secrets/quick-create-cli

File: src/azure-cli/azure/cli/command_modules/role/custom.py [hunk 3]
```diff
@@ -1302,14 +1302,10 @@ def create_service_principal_for_rbac(
     }
     if cert_file:
         logger.warning(
-            "Please copy %s to a safe place. When you run 'az login', provide the file path in the --password argument",
+            "Please copy %s to a safe place. When you run `az login`, provide the file path in the --certificate "
```
Comment (jiasli) [inline src/azure-cli/azure/cli/command_modules/role/custom.py:1305]: Changed according to https://github.com/Azure/azure-cli/pull/30091.

File: src/azure-cli/azure/cli/command_modules/role/custom.py [hunk 4]
```diff
@@ -1302,14 +1302,10 @@ def create_service_principal_for_rbac(
...
+            "argument",
             cert_file)
         result['fileWithCertAndPrivateKey'] = cert_file
-
-    login_hint = ('To log in with this service principal, run:\n'
-                  f'az login --service-principal --username {app_id} --password {password or cert_file} '
-                  f'--tenant {graph_client.tenant}')
-    logger.info(login_hint)
```
Comment (jiasli) [inline src/azure-cli/azure/cli/command_modules/role/custom.py:1312]: This hint is removed to avoid users' potential incorrect handling of the log.

**Review by evelyn-ys — APPROVED** [review]

File: src/azure-cli/azure/cli/command_modules/role/_help.py [hunk 5]
```diff
@@ -528,18 +528,27 @@
...
-    text: az ad sp create-for-rbac -n MyApp
-  - name: Create with a Contributor role assignments on specified scopes. To retrieve current subscription ID, run `az account show --query id --output tsv`.
-    text: az ad sp create-for-rbac -n MyApp --role Contributor --scopes /subscriptions/00000000-0000-0000-0000-000000000000/resourceGroups/resourceGroup1 /subscriptions/00000000-0000-0000-0000-000000000000/resourceGroups/resourceGroup2
-  - name: Create using a self-signed certificate.
-    text: az ad sp create-for-rbac --create-cert
-  - name: Create using a self-signed certificate, and store it within KeyVault.
-    text: az ad sp create-for-rbac --keyvault MyVault --cert CertName --create-cert
-  - name: Create using existing certificate in KeyVault.
-    text: az ad sp create-for-rbac --keyvault MyVault --cert CertName
+- name: Create without role assignment.
```
[the remark this answers is not in the document]
Comment (jiasli) [reply src/azure-cli/azure/cli/command_modules/role/_help.py:531]: On the contrary, the two-space indent is unnecessary.

[the remark this answers is not in the document]
Comment (bebound) [reply src/azure-cli/azure/cli/command_modules/role/_help.py:531]: YAML has a strict indent rule; my formatter uses 2 spaces.
> In YAML [block styles](https://yaml.org/spec/1.2-old/spec.html#style/block/), structure is determined by indentation. In general, indentation is defined as **a zero or more** [space](https://yaml.org/spec/1.2-old/spec.html#space//) characters at the start of a line.  -- https://yaml.org/spec/1.2-old/spec.html#id2777534

**Review by bebound — APPROVED** [review]

**Pull request merged**